sdk/eventhub/azure-eventhubs/tests/livetest/synctests/test_eventprocessor.py

```diff
@@ -433,9 +449,211 @@ def close(self):
+
+        def _create_consumer(self, consumer_group, partition_id, event_position, **kwargs):
+            return MockEventhubConsumer(**kwargs)
+
+        def get_partition_ids(self):
+            return ["0", "1"]
+
+    mock_client = MockEventHubClient()
+    current_ownerships = [ownership_ref[o] for o in ownerships]
+    om = OwnershipManager(mock_client, TEST_CONSUMER_GROUP, TEST_OWNER, None, 10, None)
+    om._initializing = True
+    to_claim_ownership = om._balance_ownership(current_ownerships, partitions)
+    assert len(to_claim_ownership) == expected_result
+
+
+@pytest.mark.parametrize(
+    "ownerships, partitions, expected_result",
+    [
+        ([], ["0", "1", "2"], 1),
+        (['ownership_active0', 'ownership_active1'], ["0", "1", "2"], 1),
+        (['ownership_active0', 'ownership_expired'], ["0", "1", "2"], 1),
+        (['ownership_active0', 'ownership_expired', 'ownership_released'], ["0", "1", "2", "3"], 1),
+        (['ownership_active0'], ["0", "1", "2", "3"], 1),
+        (['ownership_expired', 'ownership_released'], ["0", "1", "2", "3"], 1),
+        (['ownership_active0', 'ownership_active1'], ["0", "1"], 0),
+        (['ownership_active0', 'ownership_self_owned'], ["0", "1"], 1),
+    ]
+)
+def test_balance_ownership(ownerships, partitions, expected_result):
+    ownership_ref = {
+        'ownership_active0': {
+            "fully_qualified_namespace": TEST_NAMESPACE,
+            "partition_id": "0",
+            "eventhub_name": TEST_EVENTHUB,
+            "consumer_group": TEST_CONSUMER_GROUP,
+            "owner_id": "owner_0",
+            "last_modified_time": time.time()
+        },
+        'ownership_active1': {
+            "fully_qualified_namespace": TEST_NAMESPACE,
+            "partition_id": "1",
+            "eventhub_name": TEST_EVENTHUB,
+            "consumer_group": TEST_CONSUMER_GROUP,
+            "owner_id": "owner_1",
+            "last_modified_time": time.time()
+        },
+        'ownership_self_owned': {
+            "fully_qualified_namespace": TEST_NAMESPACE,
+            "partition_id": "1",
+            "eventhub_name": TEST_EVENTHUB,
+            "consumer_group": TEST_CONSUMER_GROUP,
+            "owner_id": TEST_OWNER,
+            "last_modified_time": time.time()
+        },
+        'ownership_expired': {
+            "fully_qualified_namespace": TEST_NAMESPACE,
+            "partition_id": "2",
+            "eventhub_name": TEST_EVENTHUB,
+            "consumer_group": TEST_CONSUMER_GROUP,
+            "owner_id": "owner_1",
+            "last_modified_time": time.time() - 100000
+        },
+        'ownership_released': {
+            "fully_qualified_namespace": TEST_NAMESPACE,
+            "partition_id": "3",
+            "eventhub_name": TEST_EVENTHUB,
+            "consumer_group": TEST_CONSUMER_GROUP,
+            "owner_id": "",
+            "last_modified_time": time.time()
+        }
+    }
+    class MockEventHubClient(object):
```

Comment:
It looks like this mock can be a shared object?  (I don't immediately see any dissimilarities, and the partition IDs bit can be settable at instance level if needed)

(Same thing with default ownership refs maybe? I didn't look too deeply at that, I just see a lot of similar test boilerplate)

Reply:
Thanks - yeah I'll try to clean up some of that duplication this week :)